### PR TITLE
Implemented SingleValue Scrutinizer tests using mocks

### PR DESCRIPTION
**Pull request by darecoder1999**
Updated test class & added inner class to the scrutinizer

### CODE REVIEW

File: extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java [hunk 1]
```diff
@@ -44,6 +49,24 @@
 public class SingleValueScrutinizer extends EditScrutinizer {
 
     public static final String type = "single-valued-property-added-more-than-once";
+    public static String SINGLE_VALUE_CONSTRAINT_QID = "Q19474404";
+    public static String SINGLE_VALUE_CONSTRAINT_EXCEPTION = "P2303";
+
+    class SingleValueConstraint {
```
Comment (wetneb) [inline extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java:55]: I think all exceptions can accept constraints, so how about we pull the parsing code in this class to a common superclass of all the inner classes you have been introducing?

Comment (darecoder1999) [reply extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java:55]: Sure, we can make a constraint class that will be extended by these inner classes. And check what all parameters can be added over there other than exceptions that are generally common to all property constraints.

Comment (wetneb) [reply extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java:55]: Great! Constraint status is another one.

File: extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java [hunk 2]
```diff
@@ -44,6 +46,17 @@
...
     public static final String type = "single-valued-property-added-more-than-once";
+    public static String SINGLE_VALUE_CONSTRAINT_QID = "Q19474404";
+    public static String SINGLE_BEST_VALUE_CONSTRAINT_QID = "Q52060874";
+
+    class SingleValueConstraint extends Constraint {
+        Set<EntityIdValue> constraintExceptions;
+
+        SingleValueConstraint(Statement statement) {
+            super(statement);
+            constraintExceptions = getConstraintExceptions();
```
Comment (wetneb) [inline extensions/wikidata/src/org/openrefine/wikidata/qa/scrutinizers/SingleValueScrutinizer.java:57]: You should not need to declare a new field for constraint exceptions again: you can just reuse the one from the super class. So, for all constraints which do not take any parameters, you should not need to declare any inner class: the `Constraint` class itself can be used in the scrutinizer.

**Pull request merged**